joomla/joomla-cms
# [4.2] Switch div to img and use the loading attr. [Media Manager images list]

Pull Request for Issue #36533 [Solves 1/3 reported problems, the thumbs and the streaming/paginated response are still valid... ].

### Summary of Changes
- Add and intersectionObserver so the browser won't try to fetch every image in the list view (only the ones that are visible)


### Testing Instructions
- Apply this PR and run `npm ci` or d/l the package from github
- navigate to different subfolders in the `images` folder
- Open the browser inspector, select the Network tab and then the img tab. As you scroll in the list view more images will appear both in the list view and in the network/img tab. The Network/img should have less images that the total amount of images in the list (assuming the viewport is not revealing all the images and you haven't scrolled to the end of the list)


### Actual result BEFORE applying this Pull Request

All images fetched immediately

### Expected result AFTER applying this Pull Request

Images are fetched as they enter the viewport

### Documentation Changes Required

No

Ping @laoneo @Fedik @crystalenka
